src/Shell.php

```diff
@@ -350,17 +351,18 @@ public function doRun(InputInterface $input, OutputInterface $output)
      * execution loop.
      *
      * @throws Exception if thrown via the `throw-up` command
-     *
-     * @param InputInterface $input An Input instance
      */
-    private function doInteractiveRun(InputInterface $input)
+    private function doInteractiveRun()
     {
         $this->initializeTabCompletion();
         $this->readline->readHistory();
 
-        $this->output->writeln($this->getHeader());
-        $this->writeVersionInfo();
-        $this->writeStartupMessage();
+        // Show a startup message, as long as output isn't piped.
+        if (!$this->config->outputIsPiped()) {
```

Comment:
After playing with it a bit, this isn't right. A forced interactive run should have a header, just like an implicitly interactive run. Pushing a fix for this.